scope

```diff
@@ -128,18 +148,33 @@ EOF
         ;;
 
     launch)
-        check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
-        docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
-
-        CONTAINER=$($(launch_command "$@"))
-        echo $CONTAINER
+        if ! check_docker_for_mac || check_probe_only ; then
+            check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
+            docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
+            CONTAINER=$($(launch_command "$@"))
+            echo $CONTAINER
+        elif check_docker_for_mac ; then
+            check_not_running $SCOPE_APP_CONTAINER_NAME $SCOPE_IMAGE_NAME
+            docker rm -f $SCOPE_APP_CONTAINER_NAME >/dev/null 2>&1 || true
+            CONTAINER=$($(launch_app_command "$@"))
+            echo $CONTAINER
+            app_ip=$(docker inspect -f '{{.NetworkSettings.IPAddress}}' "${CONTAINER}")
+            check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
+            docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
+            CONTAINER=$($(launch_command --no-app "${app_ip}:4040" "$@"))
```

Comment:
All comments had been addressed.

On Wed, 4 May 2016, 14:32 Tom Wilkie, <notifications@github.com> wrote:

> In scope
> <https://github.com/weaveworks/scope/pull/1414#discussion_r62038495>:
>
> > -        CONTAINER=$($(launch_command "$@"))
> > -        echo $CONTAINER
> > +        if ! check_docker_for_mac || check_probe_only ; then
> > +            check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
> > +            docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
> > +            CONTAINER=$($(launch_command "$@"))
> > +            echo $CONTAINER
> > +        elif check_docker_for_mac ; then
> > +            check_not_running $SCOPE_APP_CONTAINER_NAME $SCOPE_IMAGE_NAME
> > +            docker rm -f $SCOPE_APP_CONTAINER_NAME >/dev/null 2>&1 || true
> > +            CONTAINER=$($(launch_app_command "$@"))
> > +            echo $CONTAINER
> > +            app_ip=$(docker inspect -f '{{.NetworkSettings.IPAddress}}' "${CONTAINER}")
> > +            check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
> > +            docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
> > +            CONTAINER=$($(launch_command --no-app "${app_ip}:4040" "$@"))
>
> Yes the order matters.
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/weaveworks/scope/pull/1414/files/8fbde73c3ce291c09ed09b97ba6cfe236602c7f3#r62038495>
>